src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,47 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class type) {
+        MachineProducer<Excavator> excavatorProducer = new ExcavatorProducer();
+        MachineProducer<Bulldozer> bulldozerProducer = new BulldozerProducer();
+        MachineProducer<Truck> truckProducer = new TruckProducer();
+        List<? extends Machine> machines = new ArrayList<>();
+        if (Excavator.class.equals(type)) {
+            machines = excavatorProducer.get();
+        } else if (Bulldozer.class.equals(type)) {
+            machines = bulldozerProducer.get();
+        } else if (Truck.class.equals(type)) {
+            machines = truckProducer.get();
+        } else {
+            return Collections.emptyList();
+        }
+        return (List<Machine>) machines;
```

Comment:
You can simplify your code like this
```suggestion
        if (type == Bulldozer.class) {
            return new ArrayList<>(new BulldozerProducer().get());
        }
        if (type == Excavator.class) {
            return new ArrayList<>(new ExcavatorProducer().get());
        }
        if (type == Truck.class) {
            return new ArrayList<>(new TruckProducer().get());
        }
        return new ArrayList<>();
```